packages/edit-widgets/src/store/actions.js

```diff
@@ -103,17 +103,15 @@ export function* saveWidgetArea( widgetAreaId ) {
 
 	// Remove all duplicate reference widget instances
 	const usedReferenceWidgets = [];
-	const widgetsBlocks = post.blocks.filter(
-		( { attributes: { referenceWidgetName } } ) => {
-			if ( referenceWidgetName ) {
-				if ( usedReferenceWidgets.includes( referenceWidgetName ) ) {
-					return false;
-				}
-				usedReferenceWidgets.push( referenceWidgetName );
+	const widgetsBlocks = post.blocks.filter( ( { attributes: { id } } ) => {
+		if ( id ) {
+			if ( usedReferenceWidgets.includes( id ) ) {
+				return false;
 			}
-			return true;
+			usedReferenceWidgets.push( id );
 		}
-	);
+		return true;
+	} );
```

Comment:
Let's try to simplify these into just a `_.uniqBy` call?

Reply:
Non-`WP_Widget` widgets have no `id`, though, so `uniqBy( post.blocks, [ 'attributes', 'id' ] )` would remove every `WP_Widget` widget except for the first one.

Reply:
Ahh got it. Then probably a `Set` rather than an array for `usedReferenceWidgets` would be more readable and performant then? That's optional though just nitpicking.

Reply:
I'll leave it as is for now. I have it on my list to go through the `saveWidgetArea` method and try to simplify it. For example I think it's confusing how we're keeping track of IDs using a sparse array.